### DISCUSSION TITLE
Recommendations on how to display citations for documents in Blob Storage

### DISCUSSION THREAD

**Discussion by ianlcassidy** [microsoft/sample-app-aoai-chatGPT#652] [General]
Regarding https://github.com/microsoft/sample-app-aoai-chatGPT?tab=readme-ov-file#changing-citation-display

The original files that I am chunking are in Blob Storage, but the Citation panel is configured to disable the clickable hyperlink if the `url` returned by AI Search has `blob.core` in the url. I have received requests from users that they would like to be able to view and download the original documents (not just chunks) from the Citation panel.

Does anyone have any recommendations for workarounds for this? Fork the repo and remove this restriction? Put the original files in Sharepoint or some other file share and reference that url instead? Something else...?

**Comment by erolmam-75** [comment]
Hi @ianlcassidy , I am not a super-experienced developer but I have created a RAG solution in Azure using also this web page.
I want to utilize the Citation with link URLs to the documents. Can you describe how I can use it? Do I define the URL per document in the document metadata?
I hope you received feedback and solution to your question above...

**Reply by ianlcassidy** [reply]
Hi @erolmam-75 - I was more asking about the fact that they have [hardcoded a filter](https://github.com/microsoft/sample-app-aoai-chatGPT?tab=readme-ov-file#changing-citation-display) on blob storage urls so that they don't show up as embedded links in the citations panel.

If your urls are _not_ linking back to blob storage, then you can use the `AZURE_SEARCH_URL_COLUMN` environment variable to map your url field in your Azure AI Search schema. And, yes, I have defined a url per document in my Azure AI Search records. I don't think you should define it in a `metadata` field because the mapping wants a single field.

Hope this helps.

**Reply by erolmam-75** [reply]
Thanks @ianlcassidy for taking your time to answer me!!! 
How exactly did you "defined a url per document in your Azure AI Search records"? Can I just call my field doc_url in AI Search and place the URL link for each document? Then set AZURE_SEARCH_URL_COLUMN = doc_url?

Do you think you can describe "step by step" or refer me to some detailed description?
I have deployed this web directly from AI Studio and I am not doing any advanced programming, so you get a grip which level I am on 😊

I am also adding one of the greatest contributor for this project @sarah-widder , for support on this...

**Reply by ianlcassidy** [reply]
I have a custom AI Search index schema and I have defined the url field in there:
![image](https://github.com/user-attachments/assets/8678d936-9db9-441e-8316-dae999a98b1a)

You could add this field to your existing AI Search index and then update the records using the AI Search API or SDK. Beyond that, I don't really have a "step by step" guide.

**Reply by erolmam-75** [reply]
Thanks once again @ianlcassidy ! 
You maybe noticed that I just edited my question above, for example to set AZURE_SEARCH_URL_COLUMN = url in your case.

**Reply by ianlcassidy** [reply]
Yes, exactly. You'd go into your app service and update the environment variables:
![image](https://github.com/user-attachments/assets/15bae50e-d3a1-4e72-b029-37196b54fafe)
